server/src/main/java/org/elasticsearch/index/search/MatchQuery.java

```diff
@@ -296,8 +297,13 @@ protected final Query parseInternal(Type type, String fieldName, MatchQueryBuild
     }
 
     protected Analyzer getAnalyzer(MappedFieldType fieldType, boolean quoted) {
+        TextSearchInfo tsi = fieldType.getTextSearchInfo();
```

Comment:
This looks like it's changing behaviour, but is actually just detecting the error earlier - previously we would go through analysis and parsing and then blow up when calling one of the query factory methods on MappedFieldType.